src/main/java/com/synopsys/integration/detect/tool/detector/inspectors/nuget/OnlineNugetInspectorResolver.java

```diff
@@ -0,0 +1,66 @@
+package com.synopsys.integration.detect.tool.detector.inspectors.nuget;
+
+import java.io.File;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.synopsys.integration.detect.tool.cache.InstalledToolLocator;
+import com.synopsys.integration.detect.tool.cache.InstalledToolManager;
+import com.synopsys.integration.detect.workflow.file.DirectoryManager;
+import com.synopsys.integration.detectable.ExecutableTarget;
+import com.synopsys.integration.detectable.detectable.exception.DetectableException;
+import com.synopsys.integration.detectable.detectable.inspector.nuget.NugetInspectorResolver;
+
+public class OnlineNugetInspectorResolver implements NugetInspectorResolver {
+    private static final String INSPECTOR_NAME = "detect-nuget-inspector";
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final ArtifactoryNugetInspectorInstaller installer;
+    private final DirectoryManager directoryManager;
+    private final InstalledToolManager installedToolManager;
+    private final InstalledToolLocator installedToolLocator;
+
+    private boolean hasResolvedInspector = false;
+
+    public OnlineNugetInspectorResolver(
+        ArtifactoryNugetInspectorInstaller installer,
+        DirectoryManager directoryManager,
+        InstalledToolManager installedToolManager,
+        InstalledToolLocator installedToolLocator
+    ) {
+        this.installer = installer;
+        this.directoryManager = directoryManager;
+        this.installedToolManager = installedToolManager;
+        this.installedToolLocator = installedToolLocator;
+    }
+
+    @Override
+    public ExecutableTarget resolveNugetInspector() throws DetectableException {
+        File inspectorFile = null;
+        if (!hasResolvedInspector) {
+            hasResolvedInspector = true;
+
+            File installDirectory = directoryManager.getPermanentDirectory(INSPECTOR_NAME);
+            try {
+                inspectorFile = installer.install(installDirectory);
+            } catch (DetectableException e) {
+                logger.debug("Unable to install the detect nuget inspector from Artifactory.");
```

Comment:
We should log the exception
```suggestion
                logger.debug("Unable to install the detect nuget inspector from Artifactory.", e);
```